### PR TITLE
[Refactor] Refactor the kill command flagging process

### PR DESCRIPTION
**Pull request by Laitr0n**
Part of https://github.com/containerd/nerdctl/issues/1680

- [x]  Create a file in pkg/api/types/${cmd}_types.go, and define the CommandOption for this command
- [x] Create some file in pkg/cmd/${cmd}, and move the command entry point in real into this package

Signed-off-by: Laitron <meetlq@outlook.com>

### CODE REVIEW

File: pkg/cmd/kill/kill.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
+/*
```
Comment (AkihiroSuda) [inline pkg/cmd/kill/kill.go:1]: This should be in `pkg/cmd/container/kill.go`, as the canonical form of `nerdctl kill` is `nerdctl container kill`

Comment (AkihiroSuda) [reply pkg/cmd/kill/kill.go:1]: Yes, please

File: pkg/cmd/container/kill.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	"github.com/containerd/containerd/cio"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/containerd/nerdctl/pkg/clientutil"
+	"github.com/containerd/nerdctl/pkg/idutil/containerwalker"
+	"github.com/moby/sys/signal"
+	"github.com/sirupsen/logrus"
+)
+
+func Kill(ctx context.Context, KillCommandOptions types.KillCommandOptions, stdout, stderr io.Writer, args []string) error {
```
Comment (djdongjin) [inline pkg/cmd/container/kill.go:37]: can you change the func signature to 

```golang
func Kill(ctx context.Context, containers []string, opt types.KillCommandOptions, stdout, stderr io.Writer) error
```

`containers []string` is the direct object `Kill` applies to, `opt` decides how to `kill`, `stdout/stderr` is mostly for output (not directly related to `Kill`)

Comment (djdongjin) [reply pkg/cmd/container/kill.go:37]: also `args`-> `containers` (more straightforward), `KillCommandOptions` -> `opt` (1st char shouldn't be uppercased, `opt` is shorter and straightforward)

File: pkg/cmd/container/kill.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	status, err := task.Status(ctx)
+	if err != nil {
+		return err
+	}
+
+	paused := false
+
+	switch status.Status {
+	case containerd.Created, containerd.Stopped:
+		return fmt.Errorf("cannot kill container: %s: Container %s is not running", container.ID(), container.ID())
```
Comment (djdongjin) [inline pkg/cmd/container/kill.go:95]: remove the first `:` and duplicated id (-> `cannot kill container %s: container not running`)

File: pkg/cmd/container/kill.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+	}
+
+	if err := task.Kill(ctx, signal); err != nil {
+		return err
+	}
+
+	// signal will be sent once resume is finished
+	if paused {
+		if err := task.Resume(ctx); err != nil {
+			logrus.Warnf("Cannot unpause container %s: %s", container.ID(), err)
```
Comment (djdongjin) [inline pkg/cmd/container/kill.go:108]: Cannot -> cannot

File: pkg/cmd/container/kill.go (hunk 2 above)
Comment (djdongjin) [reply pkg/cmd/container/kill.go]: SGTM. it's more a nit than a change request :)

**Review by djdongjin — COMMENTED** [review]
LGTM, thanks

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**